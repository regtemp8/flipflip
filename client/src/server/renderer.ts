import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
