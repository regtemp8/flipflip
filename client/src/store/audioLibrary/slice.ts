import { createSlice, type PayloadAction } from '@reduxjs/toolkit'

interface AudioLibraryState {
  error: boolean
  loadingMetadata: boolean
  loadingSources: boolean
  drawerOpen: boolean
  selectedTags: string[]
  openMenu?: string
  playlistID?: number
  importURL?: string
}

export const initialAudioLibraryState: AudioLibraryState = {
  error: false,
  loadingMetadata: false,
  loadingSources: false,
  drawerOpen: false,
  selectedTags: []
}

export const audioLibrarySlice = createSlice({
  name: 'audioLibrary',
  // `createSlice` will infer the state type from the `initialState` argument
  initialState: initialAudioLibraryState,
  reducers: {
    setError: (state, action: PayloadAction<boolean>) => {
      state.error = action.payload
    },
    setLoadingMetadata: (state, action: PayloadAction<boolean>) => {
      state.loadingMetadata = action.payload
    },
    setLoadingSources: (state, action: PayloadAction<boolean>) => {
      state.loadingSources = action.payload
    },
    setOpenMenu: (state, action: PayloadAction<string | undefined>) => {
      state.openMenu = action.payload
    },
    setPlaylistID: (state, action: PayloadAction<number | undefined>) => {
      state.playlistID = action.payload
    },
    setDrawerOpen: (state, action: PayloadAction<boolean>) => {
      state.drawerOpen = action.payload
    },
    setImportURL: (state, action: PayloadAction<string | undefined>) => {
      state.importURL = action.payload
    },
    setSelectedTags: (state, action: PayloadAction<string[]>) => {
      state.selectedTags = action.payload
    },
    setAudioLibraryPlaylistID: (state, action: PayloadAction<string>) => {
      state.playlistID = Number(action.payload)
    }
  }
})

export const {
  setError,
  setLoadingMetadata,
  setLoadingSources,
  setOpenMenu,
  setPlaylistID,
  setDrawerOpen,
  setImportURL,
  setSelectedTags,
  setAudioLibraryPlaylistID
} = audioLibrarySlice.actions

export default audioLibrarySlice.reducer
