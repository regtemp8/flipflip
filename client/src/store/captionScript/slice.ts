import { createSlice, type PayloadAction } from '@reduxjs/toolkit'
import type CaptionScript from './CaptionScript'
import {
  type EntryState,
  type EntryUpdate,
  setEntrySlice,
  setEntry,
  updateEntry
} from '../EntryState'
import { type FontSettingsType } from './FontSettings'
import { getCaptionScript, getCaptionScriptFontSettings } from './selectors'

export interface FontSettingsUpdate<T> extends EntryUpdate<T> {
  type: FontSettingsType
}

export const initialCaptionScriptState: EntryState<CaptionScript> = {
  name: 'captionScriptSlice',
  nextID: 1,
  entries: {}
}
export const captionScriptSlice = createSlice({
  name: 'captionScripts',
  // `createSlice` will infer the state type from the `initialState` argument
  initialState: initialCaptionScriptState,
  reducers: {
    setCaptionScriptSlice: (
      state,
      action: PayloadAction<EntryState<CaptionScript>>
    ) => {
      setEntrySlice(state, action.payload)
    },
    setCaptionScript: (state, action: PayloadAction<CaptionScript>) => {
      setEntry(state, action.payload)
    },
    updateCaptionScript: (state, action: PayloadAction<CaptionScript>) => {
      updateEntry(state, action.payload)
    },
    setCaptionScriptScript: (
      state,
      action: PayloadAction<EntryUpdate<string>>
    ) => {
      getCaptionScript(state, action.payload.id).script = action.payload.value
    },
    setCaptionScriptURL: (
      state,
      action: PayloadAction<EntryUpdate<string>>
    ) => {
      getCaptionScript(state, action.payload.id).url = action.payload.value
    },
    setCaptionScriptStopAtEnd: (
      state,
      action: PayloadAction<EntryUpdate<boolean>>
    ) => {
      getCaptionScript(state, action.payload.id).stopAtEnd =
        action.payload.value
    },
    setCaptionScriptNextSceneAtEnd: (
      state,
      action: PayloadAction<EntryUpdate<boolean>>
    ) => {
      getCaptionScript(state, action.payload.id).nextSceneAtEnd =
        action.payload.value
    },
    setCaptionScriptSyncWithAudio: (
      state,
      action: PayloadAction<EntryUpdate<boolean>>
    ) => {
      getCaptionScript(state, action.payload.id).syncWithAudio =
        action.payload.value
    },
    setCaptionScriptOpacity: (
      state,
      action: PayloadAction<EntryUpdate<number>>
    ) => {
      getCaptionScript(state, action.payload.id).opacity = action.payload.value
    },
    setCaptionScriptFontSettingsBorder: (
      state,
      action: PayloadAction<FontSettingsUpdate<boolean>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).border = action.payload.value
    },
    setCaptionScriptFontSettingsColor: (
      state,
      action: PayloadAction<FontSettingsUpdate<string>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).color = action.payload.value
    },
    setCaptionScriptFontSettingsBorderColor: (
      state,
      action: PayloadAction<FontSettingsUpdate<string>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).borderColor = action.payload.value
    },
    setCaptionScriptFontSettingsFontFamily: (
      state,
      action: PayloadAction<FontSettingsUpdate<string>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).fontFamily = action.payload.value
    },
    setCaptionScriptFontSettingsFontSize: (
      state,
      action: PayloadAction<FontSettingsUpdate<number>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).fontSize = action.payload.value
    },
    setCaptionScriptFontSettingsBorderPx: (
      state,
      action: PayloadAction<FontSettingsUpdate<number>>
    ) => {
      getCaptionScriptFontSettings(
        state,
        action.payload.id,
        action.payload.type
      ).borderpx = action.payload.value
    },
    setCaptionScriptTags: (
      state,
      action: PayloadAction<EntryUpdate<number[]>>
    ) => {
      state.entries[action.payload.id].tags = action.payload.value
    },
    setCaptionScriptToggleTag: (
      state,
      action: PayloadAction<EntryUpdate<number>>
    ) => {
      const script = state.entries[action.payload.id]
      const index = script.tags.indexOf(action.payload.value)
      if (index === -1) {
        script.tags.push(action.payload.value)
      } else {
        script.tags.splice(index, 1)
      }
    },
    setCaptionScriptMarked: (
      state,
      action: PayloadAction<EntryUpdate<boolean>>
    ) => {
      getCaptionScript(state, action.payload.id).marked = action.payload.value
    }
  }
})

export const {
  setCaptionScriptSlice,
  setCaptionScript,
  updateCaptionScript,
  setCaptionScriptScript,
  setCaptionScriptURL,
  setCaptionScriptStopAtEnd,
  setCaptionScriptNextSceneAtEnd,
  setCaptionScriptSyncWithAudio,
  setCaptionScriptOpacity,
  setCaptionScriptFontSettingsBorder,
  setCaptionScriptFontSettingsColor,
  setCaptionScriptFontSettingsBorderColor,
  setCaptionScriptFontSettingsFontFamily,
  setCaptionScriptFontSettingsFontSize,
  setCaptionScriptFontSettingsBorderPx,
  setCaptionScriptTags,
  setCaptionScriptToggleTag,
  setCaptionScriptMarked
} = captionScriptSlice.actions

export default captionScriptSlice.reducer
