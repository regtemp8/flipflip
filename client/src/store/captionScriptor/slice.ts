import { createSlice, type PayloadAction } from '@reduxjs/toolkit'

interface CaptionScriptorState {
  sceneID: number
  captionScriptID: number
  sceneScripts: string[]
}

export const initialCaptionScriptorState: CaptionScriptorState = {
  sceneID: 0,
  captionScriptID: -1,
  sceneScripts: []
}
export const captionScriptorSlice = createSlice({
  name: 'captionScriptor',
  // `createSlice` will infer the state type from the `initialState` argument
  initialState: initialCaptionScriptorState,
  reducers: {
    setCaptionScriptorSceneID: (state, action: PayloadAction<number>) => {
      state.sceneID = action.payload
    },
    setCaptionScriptorSceneScripts: (
      state,
      action: PayloadAction<string[]>
    ) => {
      state.sceneScripts = action.payload
    },
    setCaptionScriptorCaptionScriptID: (
      state,
      action: PayloadAction<number | undefined>
    ) => {
      state.captionScriptID = action.payload ?? -1
    }
  }
})

export const {
  setCaptionScriptorSceneID,
  setCaptionScriptorSceneScripts,
  setCaptionScriptorCaptionScriptID
} = captionScriptorSlice.actions

export default captionScriptorSlice.reducer
