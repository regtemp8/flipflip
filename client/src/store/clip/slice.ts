import { createSlice, type PayloadAction } from '@reduxjs/toolkit'
import type Clip from './Clip'
import {
  type EntryState,
  type EntryUpdate,
  setEntrySlice,
  setEntry
} from '../EntryState'

export const initialClipState: EntryState<Clip> = {
  name: 'clipSlice',
  nextID: 1,
  entries: {}
}

export default function createClipReducer(clipState?: EntryState<Clip>) {
  return createClipSlice(clipState).reducer
}

function createClipSlice(clipState?: EntryState<Clip>) {
  const initialState = clipState ?? initialClipState
  return createSlice({
    name: 'clips',
    // `createSlice` will infer the state type from the `initialState` argument
    initialState,
    reducers: {
      setClipSlice: (state, action: PayloadAction<EntryState<Clip>>) => {
        setEntrySlice(state, action.payload)
      },
      setClip: (state, action: PayloadAction<Clip>) => {
        setEntry(state, action.payload)
      },
      setClips: (state, action: PayloadAction<Clip[]>) => {
        action.payload.forEach((s) => setEntry(state, s))
      },
      setClipToggleTag: (state, action: PayloadAction<EntryUpdate<number>>) => {
        const clip = state.entries[action.payload.id]
        const index = clip.tags.indexOf(action.payload.value)
        if (index === -1) {
          clip.tags.push(action.payload.value)
        } else {
          clip.tags.splice(index, 1)
        }
      },
      setClipVolume: (
        state,
        action: PayloadAction<EntryUpdate<number | undefined>>
      ) => {
        state.entries[action.payload.id].volume = action.payload.value
      },
      setClipTags: (state, action: PayloadAction<EntryUpdate<number[]>>) => {
        state.entries[action.payload.id].tags = action.payload.value
      },
      setClipStartEnd: (
        state,
        action: PayloadAction<EntryUpdate<number[]>>
      ) => {
        const clip = state.entries[action.payload.id]
        clip.start = action.payload.value[0]
        clip.end = action.payload.value[1]
      }
    }
  })
}

export const {
  setClipSlice,
  setClip,
  setClips,
  setClipToggleTag,
  setClipTags,
  setClipVolume,
  setClipStartEnd
} = createClipSlice().actions
