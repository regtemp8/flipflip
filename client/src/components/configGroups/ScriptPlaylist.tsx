/// <reference path="../../react-sortablejs.d.ts" />
import React, { MouseEvent, useState } from 'react'
import Sortable from 'react-sortablejs'

import {
  Fab,
  IconButton,
  List,
  ListItem,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
  type Theme,
  Tooltip
} from '@mui/material'

import { makeStyles } from 'tss-react/mui'

import AddIcon from '@mui/icons-material/Add'
import BuildIcon from '@mui/icons-material/Build'
import DeleteIcon from '@mui/icons-material/Delete'
import RepeatIcon from '@mui/icons-material/Repeat'
import RepeatOneIcon from '@mui/icons-material/RepeatOne'
import ShuffleIcon from '@mui/icons-material/Shuffle'

import { RP } from 'flipflip-common'
import SourceIcon from '../library/SourceIcon'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { addScript } from '../../store/app/slice'
import { selectCaptionScriptUrl } from '../../store/captionScript/selectors'
import flipflip from '../../FlipFlipService'
import { selectPlaylist } from '../../store/playlist/selectors'
import {
  setPlaylistChangeRepeat,
  setPlaylistRemoveItem,
  setPlaylistSortItems,
  setPlaylistToggleShuffle
} from '../../store/playlist/slice'
import ScriptOptions from '../library/ScriptOptions'

const useStyles = makeStyles()((theme: Theme) => ({
  scriptList: {
    paddingLeft: 0
  },
  thumb: {
    width: theme.spacing(6),
    height: theme.spacing(6)
  },
  playlistAction: {
    textAlign: 'center'
  },
  left: {
    float: 'left',
    paddingLeft: theme.spacing(2)
  },
  right: {
    float: 'right',
    paddingRight: theme.spacing(2)
  },
  scriptThumb: {
    height: 40,
    width: 40,
    overflow: 'hidden',
    display: 'flex',
    justifyContent: 'center',
    cursor: 'pointer',
    userSelect: 'none'
  },
  listAvatar: {
    width: 56
  },
  avatar: {
    backgroundColor: theme.palette.primary.main,
    boxShadow: 'none'
  },
  sourceIcon: {
    color: theme.palette.primary.contrastText
  }
}))

export interface ScriptPlaylistItemProps {
  playlistID: number
  scriptID: number
  index: number
  sceneID: number
  scripts: number[]
  onSourceOptions: (id: number) => void
}

export function ScriptPlaylistItem(props: ScriptPlaylistItemProps) {
  const { playlistID, index } = props
  const dispatch = useAppDispatch()
  const url = useAppSelector(selectCaptionScriptUrl(props.scriptID))

  const onSourceIconClick = (e: MouseEvent<HTMLDivElement>) => {
    const sourceURL = url as string
    if (e.shiftKey && !e.ctrlKey) {
      flipflip()
        .api.getFileUrl(sourceURL)
        .then((fileURL) => window.open(fileURL, '_blank')?.focus())
    } else if (!e.shiftKey && e.ctrlKey) {
      flipflip().api.showItemInFolder(sourceURL)
    } else if (!e.shiftKey && !e.ctrlKey) {
      // TODO make playScript work
      // dispatch(playScript(props.scriptID, props.sceneID, props.scripts))
    }
  }

  const removeScript = () => {
    dispatch(
      setPlaylistRemoveItem({
        id: playlistID,
        value: index
      })
    )
  }

  const { classes } = useStyles()
  return (
    <ListItem>
      <ListItemAvatar className={classes.listAvatar}>
        <Tooltip
          disableInteractive
          placement={'bottom'}
          title={
            <div>
              &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Click:
              Play Script
              <br />
              Shift+Click: Open Source
              <br />
              &nbsp;&nbsp;Ctrl+Click: Reveal File
            </div>
          }
        >
          <div onClick={onSourceIconClick} className={classes.scriptThumb}>
            <Fab size="small" className={classes.avatar}>
              <SourceIcon url={url} className={classes.sourceIcon} />
            </Fab>
          </div>
        </Tooltip>
      </ListItemAvatar>
      <ListItemText primary={url} />
      <ListItemSecondaryAction>
        <IconButton
          edge="end"
          onClick={() => props.onSourceOptions(props.scriptID)}
          size="large"
        >
          <BuildIcon />
        </IconButton>
        <IconButton edge="end" onClick={removeScript} size="large">
          <DeleteIcon color={'error'} />
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  )
}

export interface ScriptPlaylistProps {
  playlistID: number
}

function ScriptPlaylist(props: ScriptPlaylistProps) {
  const { playlistID } = props
  const dispatch = useAppDispatch()
  const playlist = useAppSelector(selectPlaylist(playlistID))

  const [sourceOptions, setSourceOptions] = useState<number>()
  const sceneID = 0
  // const [sceneID, setSceneID] = useState<number>(0)

  const onCloseSourceOptions = () => {
    setSourceOptions(undefined)
  }
  const onSourceOptions = (scriptID: number) => {
    setSourceOptions(scriptID)
  }

  const toggleShuffle = () => {
    dispatch(setPlaylistToggleShuffle(playlistID))
  }

  const changeRepeat = () => {
    dispatch(setPlaylistChangeRepeat(playlistID))
  }

  const { classes } = useStyles()
  return (
    <>
      <List disablePadding>
        <Sortable
          className={classes.scriptList}
          options={{
            animation: 150,
            easing: 'cubic-bezier(1, 0, 0, 1)'
          }}
          onChange={(order: any, sortable: any, evt: any) => {
            const { oldIndex, newIndex } = evt
            dispatch(
              setPlaylistSortItems({
                id: props.playlistID,
                value: { oldIndex, newIndex }
              })
            )
          }}
        >
          {playlist.items.map((id, index) => (
            <ScriptPlaylistItem
              playlistID={playlistID}
              scriptID={id}
              key={index}
              index={index}
              sceneID={sceneID}
              scripts={playlist.items}
              onSourceOptions={onSourceOptions}
            />
          ))}
        </Sortable>
        <div className={classes.playlistAction}>
          <div className={classes.left}>
            <Tooltip
              disableInteractive
              title={'Shuffle ' + (playlist.shuffle ? '(On)' : '(Off)')}
            >
              <IconButton onClick={toggleShuffle} size="large">
                <ShuffleIcon color={playlist.shuffle ? 'primary' : undefined} />
              </IconButton>
            </Tooltip>
            <Tooltip
              disableInteractive
              title={
                'Repeat ' +
                (playlist.repeat === RP.none
                  ? '(Off)'
                  : playlist.repeat === RP.all
                    ? '(All)'
                    : '(One)')
              }
            >
              <IconButton onClick={changeRepeat} size="large">
                {playlist.repeat === RP.none && <RepeatIcon />}
                {playlist.repeat === RP.all && <RepeatIcon color={'primary'} />}
                {playlist.repeat === RP.one && (
                  <RepeatOneIcon color={'primary'} />
                )}
              </IconButton>
            </Tooltip>
          </div>
          <Tooltip disableInteractive title="Add Tracks">
            <IconButton
              onClick={() => dispatch(addScript(props.playlistID))}
              size="large"
            >
              <AddIcon />
            </IconButton>
          </Tooltip>
        </div>
      </List>
      {sourceOptions != null && (
        <ScriptOptions scriptID={sourceOptions} onDone={onCloseSourceOptions} />
      )}
    </>
  )
}

;(ScriptPlaylist as any).displayName = 'ScriptPlaylist'
export default ScriptPlaylist
