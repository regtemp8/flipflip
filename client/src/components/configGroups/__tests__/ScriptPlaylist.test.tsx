import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import ScriptPlaylist from "../ScriptPlaylist";
import TestProvider from "../../../util/TestProvider";
import { RP } from "flipflip-common";
import { setScene, setSceneAddScriptPlaylist } from "../../../store/scene/slice";
import { setCaptionScript } from "../../../store/captionScript/slice";
import store from "../../../store/store";
import { newScene } from "../../../store/scene/Scene";

jest.mock('../../library/SourceIcon', () => 'SourceIcon');

// mocking this so that test doesn't throw error
jest.mock('react-sortablejs', () => 'Sortable');

describe("ScriptPlaylist", () => {
  it("should match snapshot", () => {
    const sceneID = 3
    const scriptID = 7
    store.dispatch(setScene(newScene({ id: sceneID })))
    store.dispatch(setCaptionScript({id: scriptID }))
    store.dispatch(setSceneAddScriptPlaylist({ id: sceneID, value: { scripts: [scriptID], shuffle: false, repeat: RP.none }}))
    const component = renderer.create(
      <TestProvider store={store}>
        <ScriptPlaylist
            sceneID={sceneID}
            playlistIndex={0}
            onSourceOptions={(scriptID) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});