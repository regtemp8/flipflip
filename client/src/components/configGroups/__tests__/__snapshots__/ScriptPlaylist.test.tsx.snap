// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptPlaylist should match snapshot 1`] = `
<ul
  className="MuiList-root css-1mk9mw3-MuiList-root"
>
  <Sortable
    className="ScriptPlaylist-scriptList-1"
    onChange={[Function]}
    options={
      Object {
        "animation": 150,
        "easing": "cubic-bezier(1, 0, 0, 1)",
      }
    }
  >
    <li
      className="MuiListItem-container css-oyakqy-MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-secondaryAction css-hcmkxp-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemAvatar-root ScriptPlaylist-listAvatar-7 css-1e9lk82-MuiListItemAvatar-root"
        >
          <div
            aria-label={null}
            aria-labelledby={null}
            className="ScriptPlaylist-scriptThumb-6"
            data-mui-internal-clone-element={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default ScriptPlaylist-avatar-8 css-12m9wit-MuiButtonBase-root-MuiFab-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <SourceIcon
                className="ScriptPlaylist-sourceIcon-9"
              />
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        />
      </div>
      <div
        className="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1pux6rn-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="BuildIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m22.7 19-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1pux6rn-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </Sortable>
  <div
    className="ScriptPlaylist-playlistAction-3"
  >
    <div
      className="ScriptPlaylist-left-4"
    >
      <button
        aria-label="Shuffle (Off)"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ShuffleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.59 9.17 5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Repeat (Off)"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RepeatIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <button
      aria-label="Add Tracks"
      aria-labelledby={null}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      className="ScriptPlaylist-right-5"
    >
      <button
        aria-label="Remove Playlist"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</ul>
`;
